RR.App/Controllers/HRIS/EmployeeBankingController.cs

```diff
@@ -121,8 +125,28 @@ public async Task<IActionResult> GetBankingDetails([FromQuery] int id)
     {
         try
         {
-            var employeeBanking = await _employeeBankingService.GetBanking(id);
-            return Ok(employeeBanking);
+            var claimsIdentity = User.Identity as ClaimsIdentity;
+            var accessTokenEmail = claimsIdentity?.FindFirst(ClaimTypes.Email)?.Value;
+            var role = claimsIdentity?.FindFirst(ClaimTypes.Role)?.Value;
+
+            if (!string.IsNullOrEmpty(accessTokenEmail))
+            {
+                if ("SuperAdmin" == role || "Admin" == role || "Talent" == role || "Journey" == role)
+                {
+                    var employeeBanking = await _employeeBankingService.GetBanking(id);
+                    return Ok(employeeBanking);
+                }
+
+                var userId = GlobalVariables.GetUserId();
+
+                if (userId == id)
+                {
+                    var employeeBanking = await _employeeBankingService.GetBanking(id);
+                    return Ok(employeeBanking);
+                }
```

Comment:
Will the getbanking function not run no matter what if there is a user id populated in the controller?

Reply:
Yeah it should, this function should be preventing users who not employee to get the data unless their id matches their own. So a user can get their own banking details. And SuperAdmin etc. will be able to see banking. I think it should only be superAdmin and admin not sure on talent and journey being allowed. @Wimmas